src/student/customImplementations/DynamicArray.java

```diff
@@ -0,0 +1,67 @@
+package student.customImplementations;
+
+import student.models.Student;
+
+public class DynamicArray {
+    private Student[] sArray;
+
+    public void add(Student student) {
+        if (sArray == null) {
+            sArray = new Student[1];
+            sArray[0] = student;
+        } else {
+            Student[] curArray = new Student[sArray.length + 1];
+            for (int i = 0; i < sArray.length; i++) {
+                curArray[i] = sArray[i];
+            }
+            curArray[sArray.length] = student;
+            sArray = curArray;
+        }
+    }
+
+    private void removeAt(int index) {
+        if (sArray != null) {
+            Student[] curArray = new Student[sArray.length - 1];
```

Comment:
decrease/ increase the size of the array  each time when add/remove operations performed, by creating a new array is an expensive operation, so I will suggest resizing array when it reaches to some capacity